sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatAsyncClient.java

```diff
@@ -262,20 +263,28 @@ public ChatThreadAsyncClient getChatThreadClient(String chatThreadId) {
     }
 
     /**
-     * Receive realtime notifications.
+     * Start receiving realtime notifications.
+     * Until {@link ChatAsyncClient#stopRealtimeNotifications()} is called,
+     * realtime notifications will be enabled and the corresponding registration will auto-renew.
+     * If there's error during registration initialization or renewal,
+     * realtime notification will be disabled and errorHandler will be called.
      * @param context the Android app context
+     * @param errorHandler error handler callback for registration failures
      * @throws RuntimeException if realtime notifications failed to start.
      */
-    public void startRealtimeNotifications(Context context) {
+    public void startRealtimeNotifications(Context context, Consumer<Throwable> errorHandler) {
         if (this.signalingClient.hasStarted()) {
             return;
         }
 
-        this.signalingClient.start(context);
+        this.signalingClient.start(context, errorHandler);
     }
 
     /**
-     * Receive realtime notifications.
+     * Start receiving realtime notifications.
+     * Until {@link ChatAsyncClient#stopRealtimeNotifications()} is called,
+     * realtime notifications will be enabled and the corresponding registration will auto-renew.
+     * If there's error during registration initialization or renewal, realtime notification will be disabled.
```

Comment:
No. This method will throw for other failures during start but won't throw for registration error. Because registration happens in a different thread created by trouter client. If we add throw in callback function, the whole app would crash when it throws. I am not sure whether this is a good behavior that make this failure to break the whole App.
Instead, it would just close the realtime notification and print some error message. But as a result, user won't know the process failed.
We keep this interface without a handler here because it is already released. Should we think about deprecate this interface? @vcolin7  @srnagar @JonathanGiles Any idea?

Reply:
Ideally, we should have included the errorHandler in this API and for renewal failures, the error handler would be called. Since this is already GA, we should add another overload with errorHandler and mark this API for deprecation and keep the current behavior until we can remove the deprecated API.

Reply:
Push notifications is not GA yet. If we think this is the way to go we still could do it, but I though this was already discussed on the chat @srnagar / @CHILIU-MSFT

Reply:
@juancamilor this is for the already GA'd `startRealtimeNotifications()` method that takes a `skypeUserToken`.

Reply:
I'll mark this API for deprecation.

Reply:
When adding a `@Deprecate` tag, remember to point to the API that should be used instead by users. See this as [an example](https://github.com/Azure/azure-sdk-for-java/blob/1e6a7274eb7ad06d1613419a7b2bf6bcb7c2d221/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java#L264).